**Integrantes:** Marvin Francisconi | Ricardo Illanes  
**Curso:** Visualización de Datos (ADY1104)  
**Caso:** StreamView Analytics

### &rarr; Carga de datasets

In [5]:
import pandas as pd

# Cargar los datasets
usuarios = pd.read_csv("StreamView_Analytics_CSV/usuarios.csv")
contenidos = pd.read_csv("StreamView_Analytics_CSV/contenidos.csv")
suscripciones = pd.read_csv("StreamView_Analytics_CSV/suscripciones.csv")
dispositivos = pd.read_csv("StreamView_Analytics_CSV/dispositivos.csv")
reproducciones = pd.read_csv("StreamView_Analytics_CSV/reproducciones.csv")
calificaciones = pd.read_csv("StreamView_Analytics_CSV/calificaciones.csv")
interacciones = pd.read_csv("StreamView_Analytics_CSV/interacciones.csv")


# Diccionario para trabajar con todos los datasets
datasets = {
    "Usuarios": usuarios,
    "Contenidos": contenidos,
    "Suscripciones": suscripciones,
    "Dispositivos": dispositivos,
    "Reproducciones": reproducciones,
    "Calificaciones": calificaciones,
    "Interacciones": interacciones
}


# Exploración inicial
for nombre, df in datasets.items():
    print("=" * 70)
    print(nombre)
    print("=" * 70)
    
    print(f"Filas: {df.shape[0]}")
    print(f"Columnas: {df.shape[1]}")
    
    print("\nColumnas:")
    print(df.columns.tolist())
    
    print("\nValores nulos:")
    print(df.isnull().sum())
    
    print("\nPrimeras 5 filas:")
    display(df.head())
    
    print()

Usuarios
Filas: 1500
Columnas: 8

Columnas:
['usuario_id', 'fecha_registro', 'pais', 'ciudad', 'edad', 'segmento_edad', 'genero', 'canal_adquisicion']

Valores nulos:
usuario_id           0
fecha_registro       0
pais                 0
ciudad               0
edad                 0
segmento_edad        0
genero               0
canal_adquisicion    0
dtype: int64

Primeras 5 filas:


,usuario_id,fecha_registro,pais,ciudad,edad,segmento_edad,genero,canal_adquisicion
0,U00001,2024-02-06,Chile,Concepción,33,25-34,Prefiere no indicar,Orgánico
1,U00002,2025-05-13,Chile,Valparaíso,63,55+,Femenino,Redes sociales
2,U00003,2023-03-17,México,Ciudad de México,32,25-34,Femenino,Referido
3,U00004,2023-01-25,España,Madrid,47,45-54,Femenino,Orgánico
4,U00005,2025-03-16,Chile,La Serena,41,35-44,Femenino,Orgánico



Contenidos
Filas: 420
Columnas: 14

Columnas:
['contenido_id', 'titulo', 'tipo_contenido', 'genero_principal', 'subgenero', 'clasificacion_edad', 'anio_estreno', 'duracion_min', 'temporadas', 'episodios', 'idioma_original', 'pais_origen', 'exclusivo', 'fecha_alta_catalogo']

Valores nulos:
contenido_id           0
titulo                 0
tipo_contenido         0
genero_principal       0
subgenero              0
clasificacion_edad     0
anio_estreno           0
duracion_min           0
temporadas             0
episodios              0
idioma_original        0
pais_origen            0
exclusivo              0
fecha_alta_catalogo    0
dtype: int64

Primeras 5 filas:


,contenido_id,titulo,tipo_contenido,genero_principal,subgenero,clasificacion_edad,anio_estreno,duracion_min,temporadas,episodios,idioma_original,pais_origen,exclusivo,fecha_alta_catalogo
0,C0001,Marea Alto 1,Serie,Comedia,Romántica,16+,2012,49,1,9,Inglés,Estados Unidos,No,2023-08-19
1,C0002,Tiempo Azul 2,Película,Documental,Sociedad,16+,2001,131,0,1,Español,España,No,2025-07-21
2,C0003,Eco Global 3,Película,Acción,Supervivencia,18+,2005,123,0,1,Inglés,Estados Unidos,Sí,2025-05-09
3,C0004,Destino Rojo 4,Película,Aventura,Exploración,13+,1998,98,0,1,Inglés,España,No,2025-04-08
4,C0005,Eco Norte 5,Documental,Drama,Histórico,13+,2000,73,0,1,Español,España,No,2023-11-24



Suscripciones
Filas: 1500
Columnas: 9

Columnas:
['suscripcion_id', 'usuario_id', 'plan', 'fecha_inicio', 'fecha_fin', 'estado', 'precio_mensual_usd', 'renovacion_automatica', 'motivo_cancelacion']

Valores nulos:
suscripcion_id              0
usuario_id                  0
plan                        0
fecha_inicio                0
fecha_fin                1085
estado                      0
precio_mensual_usd          0
renovacion_automatica       0
motivo_cancelacion       1085
dtype: int64

Primeras 5 filas:


,suscripcion_id,usuario_id,plan,fecha_inicio,fecha_fin,estado,precio_mensual_usd,renovacion_automatica,motivo_cancelacion
0,S00001,U00001,Premium,2024-02-06,NaN,Activa,13.99,Sí,NaN
1,S00002,U00002,Básico,2025-05-13,NaN,Activa,6.99,Sí,NaN
2,S00003,U00003,Básico,2023-03-17,2025-12-20,Cancelada,6.99,No,Precio
3,S00004,U00004,Estándar,2023-01-25,NaN,Activa,9.99,Sí,NaN
4,S00005,U00005,Estándar,2025-03-16,NaN,Activa,9.99,Sí,NaN



Dispositivos
Filas: 2492
Columnas: 6

Columnas:
['dispositivo_id', 'usuario_id', 'tipo_dispositivo', 'sistema_operativo', 'version_app', 'fecha_registro_dispositivo']

Valores nulos:
dispositivo_id                0
usuario_id                    0
tipo_dispositivo              0
sistema_operativo             0
version_app                   0
fecha_registro_dispositivo    0
dtype: int64

Primeras 5 filas:


,dispositivo_id,usuario_id,tipo_dispositivo,sistema_operativo,version_app,fecha_registro_dispositivo
0,D00001,U00001,Móvil,iOS,7.2,2024-09-25
1,D00002,U00002,Smart TV,Tizen,7.2,2025-10-17
2,D00003,U00003,Consola,PlayStation,7.1,2025-02-09
3,D00004,U00004,Smart TV,webOS,7.2,2025-09-17
4,D00005,U00005,Smart TV,Roku OS,7.0,2025-08-29



Reproducciones
Filas: 25000
Columnas: 11

Columnas:
['reproduccion_id', 'usuario_id', 'contenido_id', 'dispositivo_id', 'fecha_hora_inicio', 'tipo_dispositivo', 'calidad_video', 'minutos_reproducidos', 'porcentaje_completado', 'abandono_temprano', 'buffering_segundos']

Valores nulos:
reproduccion_id          0
usuario_id               0
contenido_id             0
dispositivo_id           0
fecha_hora_inicio        0
tipo_dispositivo         0
calidad_video            0
minutos_reproducidos     0
porcentaje_completado    0
abandono_temprano        0
buffering_segundos       0
dtype: int64

Primeras 5 filas:


,reproduccion_id,usuario_id,contenido_id,dispositivo_id,fecha_hora_inicio,tipo_dispositivo,calidad_video,minutos_reproducidos,porcentaje_completado,abandono_temprano,buffering_segundos
0,R000001,U00975,C0223,D01616,2025-07-02 01:02:46,Tablet,HD,61.5,67.6,No,5.6
1,R000002,U00234,C0019,D00385,2025-09-16 17:48:26,Computador,4K,65.4,64.1,No,7.8
2,R000003,U00475,C0132,D00785,2025-02-21 20:29:07,Tablet,HD,28.2,53.3,No,20.8
3,R000004,U01392,C0123,D02316,2025-08-24 13:24:47,Móvil,HD,11.6,20.4,No,4.7
4,R000005,U00582,C0105,D00964,2025-06-27 18:09:27,Computador,SD,27.2,73.6,No,0.9



Calificaciones
Filas: 3119
Columnas: 7

Columnas:
['calificacion_id', 'usuario_id', 'contenido_id', 'fecha_calificacion', 'puntuacion_1_5', 'recomendaria', 'categoria_comentario']

Valores nulos:
calificacion_id         0
usuario_id              0
contenido_id            0
fecha_calificacion      0
puntuacion_1_5          0
recomendaria            0
categoria_comentario    0
dtype: int64

Primeras 5 filas:


,calificacion_id,usuario_id,contenido_id,fecha_calificacion,puntuacion_1_5,recomendaria,categoria_comentario
0,K00001,U00975,C0363,2025-08-22 08:20:32,4,Sí,Contenido
1,K00002,U00975,C0135,2025-10-26 13:24:35,5,Sí,Historia
2,K00003,U00975,C0324,2025-04-22 20:21:42,4,Sí,Contenido
3,K00004,U00975,C0299,2025-06-07 00:40:30,4,Sí,Experiencia general
4,K00005,U00234,C0008,2025-12-23 20:16:42,4,Sí,Actuación



Interacciones
Filas: 7438
Columnas: 7

Columnas:
['interaccion_id', 'usuario_id', 'contenido_id', 'fecha_hora', 'tipo_interaccion', 'canal', 'origen_interfaz']

Valores nulos:
interaccion_id      0
usuario_id          0
contenido_id        0
fecha_hora          0
tipo_interaccion    0
canal               0
origen_interfaz     0
dtype: int64

Primeras 5 filas:


,interaccion_id,usuario_id,contenido_id,fecha_hora,tipo_interaccion,canal,origen_interfaz
0,I000001,U00975,C0324,2025-04-22 20:26:42,Clic recomendación,Web,Recomendados
1,I000002,U00975,C0299,2025-06-04 01:52:30,Compartir,Web,Búsqueda
2,I000003,U00975,C0382,2025-10-18 20:44:19,Clic recomendación,App,Inicio
3,I000004,U00975,C0363,2025-08-19 09:22:32,Me gusta,App,Inicio
4,I000005,U00975,C0135,2025-10-19 13:50:35,No me gusta,Web,Recomendados


### Se van a tomar 6 características que se consideran importantes para poder generar un problema de negocio:  
* `Usuario`: País
* `Suscripcines`: Estado
* `Reproducciones`: Dispositivo
* `Calidad`: Calidad de vídeo
* `Contenidos`: Tipo
* `Género`: Género

In [8]:
print("USUARIOS")
print(usuarios["pais"].value_counts())
print("\nSegmento de edad:")
print(usuarios["segmento_edad"].value_counts())

print("\nSUSCRIPCIONES")
print(suscripciones["plan"].value_counts())
print("\nEstado:")
print(suscripciones["estado"].value_counts())

print("\nREPRODUCCIONES")
print("\nDispositivos:")
print(reproducciones["tipo_dispositivo"].value_counts())

print("\nCalidad:")
print(reproducciones["calidad_video"].value_counts())

print("\nCONTENIDOS")
print("\nTipo:")
print(contenidos["tipo_contenido"].value_counts())

print("\nGénero:")
print(contenidos["genero_principal"].value_counts())

USUARIOS
pais
Chile        470
México       267
Perú         192
Colombia     174
Argentina    171
España       159
Ecuador       67
Name: count, dtype: int64

Segmento de edad:
segmento_edad
35-44    442
25-34    438
18-24    338
45-54    217
55+       65
Name: count, dtype: int64

SUSCRIPCIONES
plan
Estándar    653
Básico      555
Premium     292
Name: count, dtype: int64

Estado:
estado
Activa       1085
Cancelada     415
Name: count, dtype: int64

REPRODUCCIONES

Dispositivos:
tipo_dispositivo
Móvil         7641
Smart TV      7010
Computador    6290
Tablet        3183
Consola        876
Name: count, dtype: int64

Calidad:
calidad_video
Full HD    10394
HD          8861
4K          3802
SD          1943
Name: count, dtype: int64

CONTENIDOS

Tipo:
tipo_contenido
Película      198
Serie         167
Documental     55
Name: count, dtype: int64

Género:
genero_principal
Drama              70
Comedia            57
Acción             47
Thriller           43
Ciencia ficción    40
Crimen  

### Resumen corto:  

* 1.500 usuarios, principalmente de Chile, México, Perú, Colombia, Argentina y España.
* La mayor concentración está entre 25 y 44 años.
* 1.085 suscripciones activas y 415 canceladas, aproximadamente 27,7% canceladas.
* El plan más utilizado es Estándar.
* El consumo está concentrado en Móvil, Smart TV y Computador.
* La calidad más utilizada es Full HD, seguida de HD.
* El catálogo tiene principalmente Películas y Series.
* Los géneros más numerosos del catálogo son Drama, Comedia y Acción.

# 1. Problema de negocio  

StreamView Analytics necesita comprender los patrones asociados a la permanencia y cancelación de suscripciones, considerando las características de sus usuarios y su comportamiento de consumo, para apoyar decisiones orientadas a la retención y experiencia del cliente.

## Pregunta de análisis:
&rarr; **¿Qué patrones de usuarios, suscripciones y consumo se relacionan con la permanencia o cancelación de las suscripciones?**

# 2. Audiencia objetivo y propósito comunicacional.

## Audiencia objetivo:  
Equipo de gestión y toma de decisiones de StreamView Analytics.

## Propósito educacional:  
Comunicar de manera clara los principales patrones relacionados con la permanencia y cancelación de suscripciones, utilizando información sobre usuarios, suscripciones y comportamiento de consumo para facilitar la toma de decisiones.